debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java

```diff
@@ -205,21 +206,28 @@ else if ("ONLINE".equals(type)) {
             }
         });
 
+        return deduplicateLogFiles(archivedLogFiles, onlineLogFiles);
+    }
+
+    @VisibleForTesting
+    public static List<LogFile> deduplicateLogFiles(Collection<LogFile> archiveLogs, Collection<LogFile> redoLogs) {
+        final List<LogFile> logFiles = new ArrayList<>();
+
         // DBZ-3563
         // To avoid duplicate log files (ORA-01289 cannot add duplicate logfile)
-        // Remove the archive log which has the same sequence number.
-        for (LogFile redoLog : onlineLogFiles) {
-            archivedLogFiles.removeIf(f -> {
-                if (f.getSequence().equals(redoLog.getSequence())) {
-                    LOGGER.debug("Removing archive log {} with duplicate sequence {} to {}", f.getFileName(), f.getSequence(), redoLog.getFileName());
+        // Remove the archive log which has the same sequence number and redo thread number.
+        for (LogFile redoLog : redoLogs) {
+            archiveLogs.removeIf(f -> {
+                if (f.getSequence().equals(redoLog.getSequence()) && f.getThread() == redoLog.getThread()) {
```

Comment:
@Naros Does it make sense to introduce a comparison method directly on `LogFile` class?

Reply:
Used the existing `#equals` method that had the same logic.